### PR TITLE
feat: add function to add visual to dashboard

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

add function to add visual to dashboard

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy control plane with CloudFront + S3 + API gateway
  - [ ] deploy control plane within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

### CODE REVIEW

File: src/control-plane/backend/lambda/api/test/api/quicksight-visual-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { applyChangeToDashboard } from '../../common/quicksight-visual-utils';
+
+describe('SQL Builder test', () => {
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/test/api/quicksight-visual-utils.test.ts:16]: might rename it as 'QuickSight visual management'

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/test/api/quicksight-visual-utils.test.ts (hunk 1 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/test/api/quicksight-visual-utils.test.ts]: fixed

**Review by zxkane — APPROVED** [review]

File: src/control-plane/backend/lambda/api/common/quicksight-visual-utils.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
+/**
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/common/quicksight-visual-utils.ts:1]: @tyyzqmf should we move this file and all quicksight templates to `store` or `service`?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/common/quicksight-visual-utils.ts (hunk 2 above)
Comment (tyyzqmf) [reply src/control-plane/backend/lambda/api/common/quicksight-visual-utils.ts:1]: It is best to move the file to `service`, and add some quicksight SDK action in `store`.

Comment (llmin) [reply src/control-plane/backend/lambda/api/common/quicksight-visual-utils.ts]: fixed

File: src/control-plane/backend/lambda/api/common/quicksight-template/dashboard.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/common/quicksight-template/dashboard.json:1]: Let's move those templates to `service/quicksight/templates`.

**Review by zxkane — APPROVED** [review]

**Pull request merged**